### PR TITLE
Story/vspc 112

### PR DESCRIPTION
**Pull request by Sagar-k93**
The list of spaces should indicate if a space doesn't have any incoming links. There should be an icon which shows a message that the space does not have any incoming links.

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Space.java [hunk 1]
```diff
@@ -53,6 +54,9 @@
     @OneToOne(targetEntity=VSImage.class)
     @NotFound(action=NotFoundAction.IGNORE)
     private IVSImage image;
+    
+    @Transient
+    private Boolean hasIncomingLinks;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Space.java:59]: java naming conventions for booleans is that the getter would be "is..." or sometimes "has..." or "can...". This won't be possible with this variable name. In other words, this needs to be renamed along with the getter/setter ;)

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListSpacesController.java [hunk 2]
```diff
@@ -1,32 +1,45 @@
...
     public String listSpaces(Model model) {
-
-        model.addAttribute("spaces", spaceRepo.findAll());
+        
+        Iterable<Space> allSpaces = spaceRepo.findAll();
+        Iterator<Space> iterator = allSpaces.iterator();
+        while(iterator.hasNext()) {
+            Space space = iterator.next();
+            space.setHasIncomingLinks( (spaceLinkRepo.getLinkedFromSpaces(space.getId())).size() > 0 ? true : false );
+        }
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListSpacesController.java:36]: business logic that needs to happen in manager class

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java [hunk 3]
```diff
@@ -253,4 +254,14 @@ public void deleteSpaceById(String id) {
 
         return spaceLinkRepo.getLinkedFromSpaces(id);
     }
+    
+    @Override
+    public Iterable<Space> getAllSpacesWithIncomingLinkInfo(Iterable<Space> spaceList) {
+        Iterator<Space> iterator = spaceList.iterator();
+        while(iterator.hasNext()) {
+            Space space = iterator.next();
+            space.setIncomingLinks( (spaceLinkRepo.getLinkedFromSpaces(space.getId())).size() > 0 ? true : false );
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:263]: merge with line 262

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java [hunk 4]
```diff
@@ -253,4 +254,14 @@ public void deleteSpaceById(String id) {
 
         return spaceLinkRepo.getLinkedFromSpaces(id);
     }
+    
+    @Override
+    public Iterable<Space> getAllSpacesWithIncomingLinkInfo(Iterable<Space> spaceList) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:259]: either change the name or don't give it a list of spaces, but right now you say "getAllSpaces" but also give it a list of spaces. What is it all spaces or just the passed ones?

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
The icon is shown for the spaces that do have incoming links, not for the ones that don't.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java (hunk 4 above)
Comment (Sagar-k93) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:259]: Changed the name. THe spaceList was a list of all spaces. Changed that too allSpaces as well.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/ISpaceManager.java [hunk 5]
```diff
@@ -28,5 +29,7 @@
     void deleteSpaceById(String id) throws SpaceDoesNotExistException;
     
     List<SpaceLink> getIncomingLinks(String id);
+    
+    Iterable<Space> addIncomingLinkInfoToAllSpaces(Iterable<Space> allSpaces);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/ISpaceManager.java:33]: still the same issue, is it all spaces or the passed spaces (which don't have to be all of them)?

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java [hunk 6]
```diff
@@ -253,4 +254,14 @@ public void deleteSpaceById(String id) {
 
         return spaceLinkRepo.getLinkedFromSpaces(id);
     }
+    
+    @Override
+    public Iterable<Space> addIncomingLinkInfoToAllSpaces(Iterable<Space> allSpaces) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:259]: calling your list "allSpaces" is confusing, why would it have to be all?

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java (hunk 6 above)
Comment (Sagar-k93) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:259]: It is all spaces, and for each space that we display in the space list page ( which is all the spaces ), I add the boolean for if they have  incoming links or not.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/ISpaceManager.java (hunk 5 above)
Comment (Sagar-k93) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/ISpaceManager.java:33]: I am passing all the spaces to this method. That's why I named it allSpaces.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Pull request merged**